### PR TITLE
Implement getOrCreateIndex method

### PR DESCRIPTION
**Pull request by emulienfou**
Implements the `getOrCreateIndex` method, as specified in [this discussion](https://github.com/meilisearch/integration-guides/issues/2)

* Add `getOrCreateIndex` method
* Add tests
* Update Readme with examples

> Related to issue #26

### CODE REVIEW

File: tests/Endpoints/ClientTest.php [hunk 1]
```diff
@@ -141,6 +141,14 @@ public function testGetIndex()
         $this->assertNull($index->getPrimaryKey());
     }
 
+    public function testCreateOrGetIndex()
```
Comment (curquiza) [inline tests/Endpoints/ClientTest.php:144]:
```suggestion
    public function testGetOrCreateIndex()
```

File: tests/Endpoints/ClientTest.php [hunk 2]
```diff
@@ -141,6 +141,14 @@ public function testGetIndex()
...
+    {
+        $index = $this->client->getOrCreateIndex('index');
+        $this->assertInstanceOf(Index::class, $index);
+        $this->assertSame('index', $index->getUid());
+        $this->assertNull($index->getPrimaryKey());
+    }
```
Comment (curquiza) [inline tests/Endpoints/ClientTest.php:150]: Can you check the method works even if the index already exists?
You can be inspired by what's been done here in [meilisearch-python](https://github.com/meilisearch/meilisearch-python):

https://github.com/meilisearch/meilisearch-python/blob/d6153c03ee87f09100af826fb33930cf115faab1/meilisearch/tests/index/test_index.py#L60-L84

Comment (emulienfou) [reply tests/Endpoints/ClientTest.php:151]: @curquiza added a new method `testGetOrCreateWithIndexAlreadyExists`

**Comment by curquiza** [comment]
@emulienfou can you rebase please? The tests should pass now 🙂

**Comment by curquiza** [comment]
Hello @emulienfou! It looks like your rebase didn't work as expected... the commits of the previous PR are in yours now.
Did you rebase with the `git rebase master` command?
Can you try to fix that?

You can join us [in our slack community](https://slack.meilisearch.com/) if you want to chat direclty with me (I'm Clémentine).

**Comment by emulienfou** [comment]
Hi @curquiza Yes I did a `git rebase`.
I cannot register to your slack I have a message of invite limit reached when I try

**Comment by curquiza** [comment]
> Yes I did a git rebase.

Weird... a git rebase does not add commits as I can see in your history...

>  I cannot register to your slack I have a message of invite limit reached when I try

I'll let you know for the slack. It would be better if we can talk together about this weird rebase directly to solve it.

**Comment by curquiza** [comment]
@emulienfou, I've just sent you an email with an invitation link! 🙂

**Pull request closed without merging**